### ISSUE TITLE
[KucoinFutures] no price data returned when performing place_order(type=market)

### ISSUE THREAD

**Issue by RoscoeTheDog** [ccxt/ccxt#16799]
I'm just noticing that if you perform a market buy with kucoinfutures exchange, there is no pricing data returned when receiving a response back.

example:
```
symbol = 'LUNC/USDT:USDT'
type = 'market'
side = 'buy'
amount = 1
price = None
params = {}
response = self.exchange.create_order(symbol, type, side, amount, price, params=params)
```

result:
```
{'id': '63e5fbf178b9640001b03c6e', 'clientOrderId': None, 'timestamp': None, 'datetime': None, 'lastTradeTimestamp': None, 'symbol': None, 'type': None, 'side': None, 'price': None, 'amount': None, 'cost': None, 'average': None, 'filled': None, 'remaining': None, 'status': None, 'fee': None, 'trades': None, 'timeInForce': None, 'postOnly': None, 'stopPrice': None, 'triggerPrice': None, 'info': {'code': '200000', 'data': {'orderId': '63e5fbf178b9640001b03c6e'}}}
```

Manually calculating this is almost impossible due to various factors like slippage, etc. I've been thinking of trying to manually calculate and insert certain values like _price average amount filled cost_ as these are used throughout my strategies to update other variables, insert stuff into db, and determine what actions to perform next.

I thought of loading markets and reading the indexPrice to retrieve the price. From there I could determine the total cost if commission fees are known in advance (which they are). I'm not entirely sure how the index price is determine from the orderbook because when reading the orderbook and checking it against the indexPrice, they are often times different. But this could be because of slippage and delay between requests. I thought of reading from the order book, checking my requested order volume against the depth, then trying to compute the average manually before placing the order, but the orderbook structure doesn't specify whether it returns a volume representative of the contract size or spot size. This makes it hard to determine what the executed order price actually was when it was received and filled.

**Comment by kroitor** [comment]
@RoscoeTheDog we need your version numbers, your code and your verbose output (enable verbose mode):
- https://github.com/ccxt/ccxt/wiki/faq#what-is-required-to-get-help
- https://github.com/ccxt/ccxt/wiki/Manual#troubleshooting

**Comment by RoscoeTheDog** [comment]
> @RoscoeTheDog we need your version numbers, your code and your verbose output (enable verbose mode):
> 
> * https://github.com/ccxt/ccxt/wiki/faq#what-is-required-to-get-help
> * https://github.com/ccxt/ccxt/wiki/Manual#troubleshooting

my apologies, I accidentally hit a key press and prematurely opened the question. I am modifying it now.

**Comment by RoscoeTheDog** [comment]
> @RoscoeTheDog we need your version numbers, your code and your verbose output (enable verbose mode):
> 
> * https://github.com/ccxt/ccxt/wiki/faq#what-is-required-to-get-help
> * https://github.com/ccxt/ccxt/wiki/Manual#troubleshooting

Updated. This is on CCXT 2.5.9 and on Windows 10. I am just noticing there is a pip upgrade to 2.7.71 so I will try that now and see if there was any difference.

**Comment by kroitor** [comment]
@RoscoeTheDog very few exchanges will return the info upon placing the order, you have to refetch the order with fetchOrder, fetchOrders, fetchOpenOrders or fetchClosedOrders to obtain the info, cause the order might still be open, even if it's a market order. If you enable verbose mode, you will notice that the info you get is from the exchange first hand, if the exchange does not specify a closing price for your market order immediately, then CCXT will not make up that data out of thin air. You will have to issue another call after placing the order to get the order status updated.

**Comment by kroitor** [comment]
@RoscoeTheDog please, run this and paste your verbose output here:

```
symbol = 'LUNC/USDT:USDT'
type = 'market'
side = 'buy'
amount = 1
price = None
params = {}
markets = self.exchange.load_markets()
self.exchange.verbose = True  # enable verbose mode after loading the markets
response = self.exchange.create_order(symbol, type, side, amount, price, params=params)
```

**Comment by RoscoeTheDog** [comment]
```
fetch Request: kucoinfutures POST https://api-futures.kucoin.com/api/v1/orders RequestHeaders: {'KC-API-KEY-VERSION': '2', 'KC-API-KEY': '63e5ef2b2815be000190e97c', 'KC-API-TIMESTAMP': '1676017498100', 'Content-Type': 'application/json', 'KC-API-PASSPHRASE': 'HyQcX7IzZ6qE+2+Q8KgTTwJ49ikDn4nG6T4+sQMBhrc=', 'KC-API-SIGN': 'Tp5HorwPD98Jtrdnc4Ik90gIwfrWLvtiLUaPor6OSWY=', 'KC-API-PARTNER-SIGN': 'B8hNIE+VsCZLfYZ/vtdCGOYLKYwzzN0XNorThUbu6W0=', 'KC-API-PARTNER': 'ccxtfutures', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: {"clientOid":"499dc110-fb4f-45a2-bd9a-0435b1a4b4cc","side":"buy","symbol":"LUNCUSDTM","type":"market","size":1,"leverage":1}
fetch Response: kucoinfutures POST https://api-futures.kucoin.com/api/v1/orders 200 ResponseHeaders: {'Date': 'Fri, 10 Feb 2023 08:24:59 GMT', 'Content-Type': 'application/json', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'Set-Cookie': 'AWSALB=cdk+0jf8aa7dARA7VROtkHwAzR8iGHMyCEy9nG3UCoOnzRcb7a30Df0/S3lne2ej9i6n8/8KNoE/QQi7J6M9EFn+5GoEciNFcKPgr1hqXmIJXbhCN5XRZwBQGa9T; Expires=Fri, 17 Feb 2023 08:24:59 GMT; Path=/, AWSALBCORS=cdk+0jf8aa7dARA7VROtkHwAzR8iGHMyCEy9nG3UCoOnzRcb7a30Df0/S3lne2ej9i6n8/8KNoE/QQi7J6M9EFn+5GoEciNFcKPgr1hqXmIJXbhCN5XRZwBQGa9T; Expires=Fri, 17 Feb 2023 08:24:59 GMT; Path=/; SameSite=None; Secure, __cf_bm=vE088ITcKc6R8EUDt_cd6U33b6b2fWkihHijyQ7zKC0-1676017499-0-AS8WQlZDm9gMR5qK540XEOIKTdXGkzoGv3uFN3ZB5MP6prtLt2LGGQJPkbLohBf1A397O+HFRJhaOsLIiPhub5k=; path=/; expires=Fri, 10-Feb-23 08:54:59 GMT; domain=.kucoin.com; HttpOnly; Secure; SameSite=None, __cfruid=2957d65ce1303b78ef4fcc0e2d3820e7f62e3f22-1676017499; path=/; domain=.kucoin.com; HttpOnly; Secure; SameSite=None', 'Strict-Transport-Security': 'max-age=9800; includeSubDomains; preload', 'CF-Cache-Status': 'DYNAMIC', 'Server': 'cloudflare', 'CF-RAY': '7973739ab976091f-SEA', 'Content-Encoding': 'gzip'} ResponseBody: {"code":"200000","data":{"orderId":"63e5ff5bc90a310001c0cce5"}}
```

Thank you. That response makes sense and its been insightful so far !

**Comment by RoscoeTheDog** [comment]
My little test after passing the order id to the fetch_order() method yielded this, which looks to have the info I'm looking for:

```
{'id': '63e6009478b9640001baa614', 'clientOrderId': 'a9614505-35e6-4952-806b-7d2d83c76c92', 'symbol': 'LUNC/USDT:USDT', 'type': 'market', 'timeInForce': 'GTC', 'postOnly': False, 'side': 'buy', 'amount': 1.0, 'price': 0.000163049, 'stopPrice': None, 'triggerPrice': None, 'cost': 0.163049, 'filled': 1.0, 'remaining': 0.0, 'timestamp': 1676017812000, 'datetime': '2023-02-10T08:30:12.000Z', 'fee': {'currency': None, 'cost': None}, 'status': 'closed', 'info': {'id': '63e6009478b9640001baa614', 'symbol': 'LUNCUSDTM', 'type': 'market', 'side': 'buy', 'price': None, 'size': 1, 'value': '0.163049', 'dealValue': '0.163049', 'dealSize': 1, 'stp': '', 'stop': '', 'stopPriceType': '', 'stopTriggered': False, 'stopPrice': None, 'timeInForce': 'GTC', 'postOnly': False, 'hidden': False, 'iceberg': False, 'leverage': '1', 'forceHold': False, 'closeOrder': False, 'visibleSize': None, 'clientOid': 'a9614505-35e6-4952-806b-7d2d83c76c92', 'remark': None, 'tags': 'ccxtfutures', 'isActive': False, 'cancelExist': False, 'createdAt': 1676017812000, 'updatedAt': 1676017812000, 'endAt': None, 'orderTime': 1676017812365043863, 'settleCurrency': 'USDT', 'status': 'done', 'filledSize': 1, 'filledValue': '0.163049', 'reduceOnly': False}, 'lastTradeTimestamp': None, 'average': 0.000163049, 'trades': [], 'fees': [{'currency': None, 'cost': None}], 'reduceOnly': None}
```

Consider this issue solved! Thank you kindly. Sorry if I caused any disruptions to your bigger picture work.

**Comment by kroitor** [comment]
@RoscoeTheDog sure, if anything, try this next:

```Python
symbol = 'LUNC/USDT:USDT'
type = 'market'
side = 'buy'
amount = 1
price = None
params = {}

markets = self.exchange.load_markets()
self.exchange.verbose = True  # enable verbose mode after loading the markets

response = self.exchange.create_order(symbol, type, side, amount, price, params=params)

# fetch the order status after placing it
updated_order = self.exchange.fetch_order(response['id'], response['symbol'])
```
